File: frontend/bookflow-web/src/app/models/book.ts
export interface Book {
  id?: string;
  title: string;
  author: string;
  genre: string;
  publishedYear: number;
  availableCopies: number;
  createdAtUtc?: string;
}

export interface PagedResult<T> {
  items: T[];
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

File: backend/BookFlow.Api/Models/Book.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace BookFlow.Api.Models;

public class Book
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    [BsonElement("title")]
    public string Title { get; set; } = string.Empty;

    [BsonElement("author")]
    public string Author { get; set; } = string.Empty;

    [BsonElement("genre")]
    public string Genre { get; set; } = string.Empty;

    [BsonElement("publishedYear")]
    public int PublishedYear { get; set; }

    [BsonElement("availableCopies")]
    public int AvailableCopies { get; set; }

    [BsonElement("createdAtUtc")]
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
}